templates/flavors/nvidia-gpu/machine-deployment.yaml

```diff
@@ -0,0 +1,204 @@
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: ${WORKER_MACHINE_COUNT}
+  selector:
+    matchLabels:
+  template:
+    spec:
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION}"
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md-0"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+        name: "${CLUSTER_NAME}-md-0"
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: AzureMachineTemplate
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      location: ${AZURE_LOCATION}
+      vmSize: ${AZURE_NODE_MACHINE_TYPE}
+      osDisk:
+        osType: "Linux"
+        diskSizeGB: 128
+        managedDisk:
+          storageAccountType: "Premium_LRS"
+      sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfigTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      useExperimentalRetryJoin: true
+      preKubeadmCommands:
+        # Enable NVIDIA container runtime package repository
+        - curl -sL https://nvidia.github.io/nvidia-container-runtime/gpgkey | apt-key add -
+        - curl -sL https://nvidia.github.io/nvidia-container-runtime/$(. /etc/os-release;echo $ID$VERSION_ID)/nvidia-container-runtime.list | tee /etc/apt/sources.list.d/nvidia-container-runtime.list
+        - apt update
+        # Install NVIDIA drivers
+        - apt install ubuntu-drivers-common -y
+        - ubuntu-drivers install
+        # Install NVIDIA container runtime
+        - apt install nvidia-container-runtime -y
+      postKubeadmCommands:
+        # Configure containerd for NVIDIA container runtime
+        - cp /etc/containerd/config.toml /etc/containerd/config.toml.old
+        - mv /etc/containerd/nvidia-config.toml /etc/containerd/config.toml
+        - systemctl restart containerd
+      joinConfiguration:
+        nodeRegistration:
+          name: '{{ ds.meta_data["local_hostname"] }}'
+          kubeletExtraArgs:
+            cloud-provider: azure
+            cloud-config: /etc/kubernetes/azure.json
+      files:
+      - path: /etc/kubernetes/azure.json
+        owner: root:root
+        permissions: "0644"
+        contentFrom:
+          secret:
+            name: ${CLUSTER_NAME}-md-0-azure-json
+            key: worker-node-azure.json
+      - path: /etc/containerd/nvidia-config.toml
```

Comment:
was this file taken from somewhere?

Reply:
Yes, from @sozercan's gist at https://gist.github.com/sozercan/51a569cf173ef7e57a375978af8edf26 which he linked to in #426. Not sure if it has other origins.